lnrpc/routerrpc/router.proto

```diff
@@ -125,8 +124,27 @@ message PaymentStatus {
     The taken route when state is SUCCEEDED.
     */
     lnrpc.Route route = 3;
+
+    /**
+    List of all shard statuses to this payment_hash [EXPERIMENTAL].
+    */
+    repeated ShardStatus shards = 4 [json_name = "shards"];
 }
 
+message ShardStatus {
+    /// Current state the shard is in.
+    PaymentState state = 1;
```

Comment:
I don't think this `PaymentState` can be the same as the payment-level state. For example `FAILED_NO_ROUTE` and `FAILED_TIMEOUT` do not apply to a shard.